backend/src/main/java/com/example/dontwastetomuch/game/GameService.java

```diff
@@ -24,19 +24,25 @@ public List<Game> getAllGames() {
         return gameRepo.findAll();
     }
 
-    public Game getOneGame(String gameId) {
+    public Game getOneOfMyGames(String gameId) {
         return gameRepo.findById(gameId).orElseThrow();
     }
 
 
-    public void switchStatus(Game game) {
-        if (game.isApproved()) {
-            game.setApproved(false);
-            gameRepo.save(game);
-        }else {
-            game.setApproved(true);
-            gameRepo.save(game);
+    public void switchStatus(Game game, MyUser user) {
+        if (!user.getRoles().stream().anyMatch(roles -> roles.contains("admin"))){
+            throw new IllegalArgumentException("No admin logged in");
```

Comment:
You can handle restrictions for diffrent roles in the SecurityConfig. As long as you wont call this method in any unrestricted context, you can save this if-statement.